phpBB/includes/functions.php

```diff
@@ -4403,8 +4403,7 @@ function phpbb_generate_debug_output(\phpbb\db\driver\driver_interface $db, \php
 */
 function page_footer($run_cron = true, $display_template = true, $exit_handler = true)
 {
-	global $db, $config, $template, $user, $auth, $cache, $phpEx;
-	global $request, $phpbb_dispatcher, $phpbb_admin_path;
+	global $phpbb_dispatcher, $phpbb_container, $template;
```

Comment:
Why is the entire function not moved to the controller helper, or preferably even a different file within the controller namespace such as `phpb\controller\functions` and called from the `$helper->render()` function. Other 'old functions' that still call `page_footer()` can just call that object & function instead.
As we are going to move more and more to controllers, so will be getting more and more functions for the controller aswell. So moving them to a separate file will help with maintainability.

Reply:
I haven't moved the entire function as this is for 3.3 and would otherwise result in a BC break. IMHO this function should be removed / deprecated in master.